el-patch.el

```diff
@@ -536,38 +547,78 @@ PATCH-DEFINITION is an unquoted list starting with `defun',
+                                        (length (nth 2 orig-def)))))
+                          'advice))
+             (register-patch
+              `(let ((table (or (bound-and-true-p el-patch--patches)
+                                (make-hash-table :test #'eq))))
+                 (setq el-patch--patches table)
+                 (setq table
+                       (puthash ',name
+                                (gethash
+                                 ',name table
+                                 (make-hash-table :test #'eq))
+                                table))
+                 (setq table
+                       (puthash ',type
+                                (gethash
+                                 ',type table
+                                 (make-hash-table :test #'equal))
+                                table))
+                 (puthash (cons ,(when advise `(quote ,advise))
+                                el-patch-variant)
+                          ',patch-definition table))))
         ;; If we need to validate the patch, then we also need to
         ;; register it at compile-time, not just at runtime.
         (when (and el-patch-validate-during-compile byte-compile-current-file)
           (eval register-patch t)
-          (el-patch-validate name type 'nomsg nil el-patch-variant))
+          (el-patch-validate name type 'nomsg nil
+                             (cons advise el-patch-variant)))
         `(progn
            ;; Register the patch in our hash. We want to do this right
            ;; away so that if there is an error then at least the user
            ;; can undo the patch (as long as it is not too terribly
            ;; wrong).
            ,register-patch
            ;; Now we actually overwrite the current definition.
-           (el-patch--stealthy-eval
-            ,definition
-            "This function was patched by `el-patch'."))))))
+           ,(if advise
+                ;; Use advice system
+                (let ((advice-name (el-patch--advice-name name
+                                                          el-patch-variant)))
+                  `(progn
+                     (el-patch--stealthy-eval
+                      ,(append
+                        (list (car definition) ;; Same type
+                              advice-name)     ;; Different name
+                        ;; Rest is the same
+                        (cddr definition))
+                      ,(format
+                        ;; The new line before the name is to avoid
+                        ;; long doc strings
+                        "This advice was defined by `el-patch' for\n`%S'."
+                        name))
+                     (advice-add (quote ,name)
+                                 :override (quote ,advice-name))))
+              `(el-patch--stealthy-eval
+                ,definition
+                "This function was patched by `el-patch'.")))))))
```

Comment:
Just realized that we probably should change "function" to something else for variables or otherwise.

Reply:
Yeah, for sure. That was an oversight on my part. We don't have to solve that now, but you can if you like. It's nontrivial since forms like `define-minor-mode` can define both a function and a variable. It's probably simplest to just say:
```suggestion
                "This definition was patched by `el-patch'.")))))))
```
for all of them.